### PR TITLE
feat: Image Viewer

### PR DESCRIPTION
**Pull request by ShGKme**
### ☑️ Resolves

* Fix: https://github.com/nextcloud/talk-desktop/issues/255

### 🖼️ Screenshots

image

image

image

### Todo

- [x] Add viewer and mimic to OCA.Viewer
- [x] Add Image Viewer
- [x] Add Video Viewer
- [x] Check group chats
- [x] Check open for guests public chats
- [x] Check 1-1
- [x] Check viewer overlay

### CODE REVIEW

**Comment by ShGKme** [comment]
Closed in flavor of https://github.com/nextcloud/talk-desktop/issues/323

**Comment by ShGKme** [comment]
Thought https://github.com/nextcloud/talk-desktop/issues/323 is important, this is quite annoying to have this feature, so we'll start with a direct solution for Viewer first.

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@
   - Untrusted certificate on Linux ([#23](https://github.com/nextcloud/talk-desktop/issues/23))
   - Dark/light theme ([#17](https://github.com/nextcloud/talk-desktop/issues/17))
 - Works with limitations:
-  - File viewer - opened in the default web-browser
+  - File viewer—only images and videos
```
Comment (Antreesy) [inline README.md:22]:
```suggestion
  - File viewer — only images and videos
```

File: src/talk/renderer/Viewer/Viewer.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		if (!Viewer._viewer) {
+			return
+		}
+		Viewer._viewer.close()
+	},
+
+	mount() {
+		const container = document.createElement('div')
+		container.id = 'viewer'
+		document.body.appendChild(container)
```
Comment (Antreesy) [inline src/talk/renderer/Viewer/Viewer.js:77]: Check for fullscreen mode

File: src/talk/renderer/Viewer/ViewerApp.vue [hunk 3]
```diff
@@ -0,0 +1,103 @@
+<!--
+  - @copyright Copyright (c) 2023 Grigorii Shartsev <me@shgk.me>
```
Comment (Antreesy) [inline src/talk/renderer/Viewer/ViewerApp.vue:2]:
```suggestion
  - @copyright Copyright (c) 2024 Grigorii Shartsev <me@shgk.me>
```

**Review by Antreesy — COMMENTED** [review]
Clever) Will test asap

File: src/talk/renderer/Viewer/ViewerApp.vue [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<template>
+	<NcModal v-if="file"
+		:show.sync="isOpen"
+		class="viewer-modal"
+		:title="file.basename"
```
Comment (Antreesy) [inline src/talk/renderer/Viewer/ViewerApp.vue:26]:
```suggestion
		:name="file.basename"
```

File: src/talk/renderer/Viewer/ViewerHandlerImages.vue [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<template>
+	<div class="media-wrapper">
+		<img v-if="file"
```
Comment (Antreesy) [inline src/talk/renderer/Viewer/ViewerHandlerImages.vue:24]: Do we need an additional `v-if` here? Seems to be already checked twice in parent component

File: src/talk/renderer/Viewer/ViewerApp.vue [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+			this.onClose()
+			this.onClose = noop
+		},
+	},
+}
+</script>
+
+<style scoped>
+.viewer-modal :deep(.modal-container) {
```
Comment (Antreesy) [inline src/talk/renderer/Viewer/ViewerApp.vue:100]: To show RightSidebar (need to be applied to the div with class `modal-mask`):
`width: calc(100vw - clamp(300px, 27vw, 500px));`

**Review by Antreesy — COMMENTED** [review]
Rest is working as expected. Not sure if it would be easy to restrict Modal width to show right sidebar, like web does it

File: src/talk/renderer/Viewer/ViewerApp.vue (hunk 6 above)
Comment (ShGKme) [reply src/talk/renderer/Viewer/ViewerApp.vue:100]: Maybe we don't need sidebar here...

Comment (Antreesy) [reply src/talk/renderer/Viewer/ViewerApp.vue:100]: Then it would block a chat during the call

Comment (ShGKme) [reply src/talk/renderer/Viewer/ViewerApp.vue:100]: But you cannot open an editable document anyway

Comment (Antreesy) [reply src/talk/renderer/Viewer/ViewerApp.vue:100]: Let's keep it as-is then and see, if someone will complain later

File: src/talk/renderer/Viewer/Viewer.js (hunk 2 above)
Comment (ShGKme) [reply src/talk/renderer/Viewer/Viewer.js]: Fixed in a different way to support original reparentViewer from Talk:
- Mount viewer when the app is initialized, so one nextTick is enought
- add `id="viewer"`

**Review by Antreesy — APPROVED** [review]
Tested with images, videos, md files (opened in browser)
As I see, fullscreen is handled by spreed code
Nice work!

**Review by Antreesy — APPROVED** [review]
Tested with images, videos, md files (opened in browser)
As I see, fullscreen is handled by spreed code
Nice work!

**Review by DorraJaouad — APPROVED** [review]
Tested ✅

**Comment by ShGKme** [comment]
Squashed

**Pull request merged**